File: reni-backend/src/main/java/pe/gob/mimp/reni/service/PerfilService.java
package pe.gob.mimp.reni.service;

import java.util.List;

import pe.gob.mimp.reni.dto.request.PerfilEliminarRequest;
import pe.gob.mimp.reni.dto.request.PerfilFuncionalidadListarRequest;
import pe.gob.mimp.reni.dto.request.PerfilFuncionalidadPerfListarRequest;
import pe.gob.mimp.reni.dto.request.PerfilListarRequest;
import pe.gob.mimp.reni.dto.request.PerfilModificarRequest;
import pe.gob.mimp.reni.dto.request.PerfilRegistrarRequest;
import pe.gob.mimp.reni.dto.response.FileResponse;
import pe.gob.mimp.reni.dto.response.OutResponse;
import pe.gob.mimp.reni.dto.response.PerfilFuncionalidadListarResponse;
import pe.gob.mimp.reni.dto.response.PerfilFuncionalidadPerfListarResponse;
import pe.gob.mimp.reni.dto.response.PerfilListarResponse;
import pe.gob.mimp.reni.dto.response.PerfilRegistrarResponse;

public interface PerfilService {

	public OutResponse<List<PerfilListarResponse>> listarPerfil(PerfilListarRequest req);

	public OutResponse<PerfilRegistrarResponse> registrarPerfil(PerfilRegistrarRequest req);

	public OutResponse<?> modificarPerfil(PerfilModificarRequest req);

	public OutResponse<?> eliminarPerfil(PerfilEliminarRequest req);

	public OutResponse<List<PerfilFuncionalidadListarResponse>> listarFuncionalidad(
			PerfilFuncionalidadListarRequest req);

	public OutResponse<List<PerfilFuncionalidadPerfListarResponse>> listarFuncionalidadPerfil(
			PerfilFuncionalidadPerfListarRequest req);
	
	public OutResponse<FileResponse> exportarListaPerfilXlsx(PerfilListarRequest c);
}

File: reni-frontend/src/main/reni-frontend/src/app/modules/intranet/services/perfil.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { OutResponse } from '../../sesion/dto/response/out.response';
import { PerfilEliminarRequest } from '../dto/request/perfil-eliminar.request';
import { PerfilFuncionalidadListarRequest } from '../dto/request/perfil-funcionalidad-listar.request';
import { PerfilFuncionalidadPerfListarRequest } from '../dto/request/perfil-funcionalidad-perf-listar.request';
import { PerfilListarRequest } from '../dto/request/perfil-listar.request';
import { PerfilModificarRequest } from '../dto/request/perfil-modificar.request';
import { PerfilRegistrarRequest } from '../dto/request/perfil-registrar.request';
import { FileResponse } from '../dto/response/file.response';
import { PerfilFuncionalidadListarResponse } from '../dto/response/perfil-funcionalidad-listar.response';
import { PerfilFuncionalidadPerfListarResponse } from '../dto/response/perfil-funcionalidad-perf-listar.response';
import { PerfilListarResponse } from '../dto/response/perfil-listar.response';
import { PerfilRegistrarResponse } from '../dto/response/perfil-registrar.response';

@Injectable()
export class PerfilService {

  constructor(private http: HttpClient) { }

  public listarPerfil(req: PerfilListarRequest): Observable<OutResponse<PerfilListarResponse[]>> {
    return this.http.post<OutResponse<PerfilListarResponse[]>>(`${environment.reniBackendUrl}/perfil/listarPerfil`, req);
  }

  public registrarPerfil(req: PerfilRegistrarRequest): Observable<OutResponse<PerfilRegistrarResponse>> {
    return this.http.post<OutResponse<PerfilRegistrarResponse>>(`${environment.reniBackendUrl}/perfil/registrarPerfil`, req);
  }

  public modificarPerfil(req: PerfilModificarRequest): Observable<OutResponse<any>> {
    return this.http.post<OutResponse<any>>(`${environment.reniBackendUrl}/perfil/modificarPerfil`, req);
  }

  public eliminarPerfil(req: PerfilEliminarRequest): Observable<OutResponse<any>> {
    return this.http.post<OutResponse<any>>(`${environment.reniBackendUrl}/perfil/eliminarPerfil`, req);
  }

  public listarFuncionalidad(req: PerfilFuncionalidadListarRequest): Observable<OutResponse<PerfilFuncionalidadListarResponse[]>> {
    return this.http.post<OutResponse<PerfilFuncionalidadListarResponse[]>>(`${environment.reniBackendUrl}/perfil/listarFuncionalidad`, req);
  }

  public listarFuncionalidadPerfil(req: PerfilFuncionalidadPerfListarRequest): Observable<OutResponse<PerfilFuncionalidadPerfListarResponse[]>> {
    return this.http.post<OutResponse<PerfilFuncionalidadPerfListarResponse[]>>(`${environment.reniBackendUrl}/perfil/listarFuncionalidadPerfil`, req);
  }

  public exportarListaPerfilXlsx(req: PerfilListarRequest): Observable<OutResponse<FileResponse>> {
    return this.http.post<OutResponse<FileResponse>>(`${environment.reniBackendUrl}/perfil/exportarListaPerfilXlsx`, req);
  }
}
